File: ClassEval_T/cpp/solution/SplitSentence.cpp

#include <boost/regex.hpp>
#include <sstream>
#include <algorithm>
#include <cctype>

#include <string>
#include <vector>

class SplitSentence {
public:
    std::vector<std::string> split_sentences(const std::string& sentences_string);
    int count_words(const std::string& sentence);
    int process_text_file(const std::string& sentences_string);
};


std::vector<std::string> SplitSentence::split_sentences(const std::string& sentences_string) {
    std::vector<std::string> sentences;
    boost::regex regex(R"((?<!\w\.\w.)(?<![A-Z][a-z]\.)(?<=\.|\?)\s)");
    boost::sregex_token_iterator iter(sentences_string.begin(), sentences_string.end(), regex, -1);
    boost::sregex_token_iterator end;

    for (; iter != end; ++iter) {
        std::string sentence = iter->str();
        
        if (!sentence.empty()) {
            
            sentence.erase(std::find_if(sentence.rbegin(), sentence.rend(), [](char ch) {
                return !std::isspace(ch);
                }).base(), sentence.end());
            if (!sentence.empty()) {
                sentences.push_back(sentence);
            }
        }
    }

    return sentences;
}


int SplitSentence::count_words(const std::string& sentence) {
    std::string cleaned_sentence;
    std::remove_copy_if(sentence.begin(), sentence.end(), std::back_inserter(cleaned_sentence),
        [](char c) { return !std::isalpha(c) && !std::isspace(c); });

    std::istringstream stream(cleaned_sentence);
    std::string word;
    int count = 0;

    while (stream >> word) {
        ++count;
    }

    return count;
}


int SplitSentence::process_text_file(const std::string& sentences_string) {
    std::vector<std::string> sentences = split_sentences(sentences_string);
    int max_count = 0;

    for (const auto& sentence : sentences) {
        int count = count_words(sentence);
        if (count > max_count) {
            max_count = count;
        }
    }

    return max_count;
}

File: BatchTestTool/java/java automated tester/Test/compile_passed/solutions/SplitSentence.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class SplitSentence {

    public String[] splitSentences(String sentencesString) {
        String regex = "(?<!\\w\\.\\w.)(?<![A-Z][a-z]\\.)[.!?]+\\s";
        String[] sentences = sentencesString.split(regex);
        return sentences;
    }

    public int countWords(String sentence) {
        String regex = "[^a-zA-Z\\s]";
        sentence = sentence.replaceAll(regex, "");
        String[] words = sentence.split("\\s+");
        return words.length;
    }

    public int processTextFile(String sentencesString) {
        String[] sentences = splitSentences(sentencesString);
        int maxCount = 0;
        for (String sentence : sentences) {
            int count = countWords(sentence);
            if (count > maxCount) {
                maxCount = count;
            }
        }
        return maxCount;
    }
}